sycl/source/detail/graph_impl.hpp

```diff
@@ -695,11 +695,17 @@ class graph_impl {
 
   // Returns the number of nodes in the Graph
   // @return Number of nodes in the Graph
-  size_t getNumberOfNodes() const {
+  size_t getNumberOfNodes() {
     size_t NumberOfNodes = 0;
-    for (const auto &Node : MRoots) {
-      NumberOfNodes += Node->depthSearchCount();
-    }
+    auto CountFunc = [&](std::shared_ptr<node_impl> &Node,
+                         std::deque<std::shared_ptr<node_impl>> &NodeStack) {
```

Comment:
```suggestion
                         std::deque<std::shared_ptr<node_impl>> &) {
```
I think this might throw up an error when run with `-werror` if left in and unused.